### PR TITLE
Add `deno-fmt` git filter

### PR DESCRIPTION
**Pull request by wojpawlik**
Works on Linux/npm, doesn't work on Windows/npm, haven't tested on yarn berry and pnpm.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# format on staging
+*.js	filter=deno-fmt
```
Comment (kt3k) [inline .gitattributes:2]: Does this file need to be copied to the user's project?

**Comment by kt3k** [comment]
Thanks for the PR.

But I think installing the git filter by default is not a good idea because people usually don't expect it and some people might already have different settings (for example, prettier). I'm happy to include these kinds of utility if it's optional and user can choose to install it.

File: .gitattributes (hunk 1 above)
Comment (wojpawlik) [reply .gitattributes:2]: The filter won't be used without it.

**Comment by kt3k** [comment]
> To clarify, this only registers the filter in .git/config. A project needs .gitattributes to opt into actually running it.

Ok. This is already opt-in

File: package.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
     "deno-bin": "./bin/deno"
   },
   "scripts": {
+    "postinstall": "git config filter.deno-fmt.clean \"$PWD/bin/deno fmt -\" || echo Failed to set up deno-fmt git filter",
```
Comment (kt3k) [inline package.json:15]: Doesn't this cause an install error on windows?

**Comment by kt3k** [comment]
How about having another bin script (like `setup-deno-fmt`), which sets up `filter.deno-fmt.clean` and add entry of `*.js	filter=deno-fmt` in `.gitattributes`. The user can invoke it like `npx setup-deno-fmt` if they want.

It doesn't feel like a good idea to modify user's config without their consent.

**Comment by wojpawlik** [comment]
That'd hinder ensuring that all committed code is formatted, defeating the point

**Comment by kt3k** [comment]
> That'd hinder ensuring that all committed code is formatted

Do these settings target this repository or all user projects of `deno-bin` package?

**Comment by wojpawlik** [comment]
All repositories containing appropriate `.gitattributes`

**Comment by kt3k** [comment]
Have you seen any repository doing a similar thing?

I think in npm ecosystem people usually set up those kinds of settings by `husky` + `lint-staged`, not directly by `postintall` script of formatter module itself. It feels intrusive to me for this module to write to the users' git config without any consent

**Comment by wojpawlik** [comment]
> Have you seen any repository doing a similar thing?

Nope, so I wanted to be the first :)

On 2nd thought, it's probably not worth it. It's difficult to document succintly (due to obscurity of filters), and it's only marginally better at handling partial stages.

**Pull request closed without merging**